### PR TITLE
Address some warnings found by static code analysis

### PR DESCRIPTION
**Pull request by Glease**

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_HeatExchanger.java [hunk 1]
```diff
@@ -207,7 +214,7 @@ public boolean onRunningTick(ItemStack aStack) {
 
                 if (superheated) tGeneratedEU /= 2; // We produce half as much superheated steam if necessary
 
-                int distilledConsumed = useWater(tGeneratedEU / 160f); // how much distilled water to consume
+                int distilledConsumed = useWater(tGeneratedEU); // how much distilled water to consume
```
[the remark this answers is not in the document]
Comment (Glease) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_HeatExchanger.java:217]: Yes.

File: src/main/java/gregtech/common/tileentities/generators/GT_MetaTileEntity_MagicalEnergyAbsorber.java [hunk 2]
```diff
@@ -388,21 +388,21 @@ private long absorbFromEnchantedItems() {
             if ((Integer) e.getKey() < Enchantment.enchantmentsList.length) {
                 Enchantment tEnchantment = Enchantment.enchantmentsList[(Integer) e.getKey()];
                 Integer tLevel = (Integer) e.getValue();
-                tEU += 1000000 * tLevel / (tEnchantment.getMaxLevel() * tEnchantment.getWeight());
+                tEU += 1000000L * tLevel / tEnchantment.getMaxLevel() / tEnchantment.getWeight();
```
[the remark this answers is not in the document]
Comment (Glease) [reply src/main/java/gregtech/common/tileentities/generators/GT_MetaTileEntity_MagicalEnergyAbsorber.java:391]: Personal preference, but yeah, it's identical. This is included in the commit by accident. The real warning is below, where there is an assignment to tStack that would have no effect, leading to bugs.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_HeatExchanger.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (leagris) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_HeatExchanger.java:217]: This is a significant functional change that causes the boiler to use 160 times more water.

[the remark this answers is not in the document]
Comment (Glease) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_HeatExchanger.java:217]: Read the whole commit. The division happens elsewhere.

**Review by boubou19 — APPROVED** [review]

**Pull request merged**